semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java

```diff
@@ -787,6 +811,30 @@
   public static final AttributeKey<String> RPC_JSONRPC_ERROR_MESSAGE =
       stringKey("rpc.jsonrpc.error_message");
 
+  /** Whether this is a received or sent message. */
+  public static final AttributeKey<String> MESSAGE_TYPE = stringKey("message.type");
+
+  /**
+   * MUST be calculated as two different counters starting from {@code 1} one for sent messages and
+   * one for received message.
+   *
+   * <p>Notes:
+   *
+   * <ul>
+   *   <li>This way we guarantee that the values will be consistent between different
+   *       implementations.
+   * </ul>
+   */
+  public static final AttributeKey<Long> MESSAGE_ID = longKey("message.id");
```

Comment:
Interesting that this was not present previously.

Reply:
Looks like it was added here:  https://github.com/open-telemetry/opentelemetry-specification/pull/1843. By you. :)

Reply:
Oh, this is for RPC! Didn't realize... Yeah we should definitely change that name 😄 

https://github.com/open-telemetry/opentelemetry-specification/pull/1914#discussion_r708388214

Reply:
Hmm, the id of the group is `rpc.message`, not just message. I thought `id` instead of `prefix` would be used for the name. But I guess that would lead to ugly identifiers in some cases where we split client/server into different groups. On the other hand, these identifiers may even be helpful.

Please consider that (well) before declaring the semconv artifact stable.

Reply:
It may make sense to add the group ID to the generated javadoc, otherwise this documentation here is misleading, since the RPC context is completely missing.

Reply:
Is the group information even available to the templates at the moment? I poked around and tried various things but couldn't sort it out. It's very likely I missed it somewhere, though.